instrumentation/apache-httpclient-4.0/build.gradle.kts

```diff
@@ -24,6 +24,7 @@ muzzle {
         group = "io.dropwizard"
         module = "dropwizard-client"
         versions = "(,)"
```

Comment:
We should limit it here. 
```
versions = "(,4.0.0)"
```

Reply:
And add support for newer versions if required. Like in netty-4.0, 4.1. But for now we should limit our support till 4.0.0

Reply:
I initially tried this.

What ends up happening is that 4.0.0 actually passes the above test. The `assertInverse=true` expects it to fail. 
```
> Task :instrumentation:apache-httpclient-4.0:muzzle-AssertFail-io.dropwizard-dropwizard-client-4.0.0 FAILED
MUZZLE PASSED io.opentelemetry.javaagent.instrumentation.hypertrace.java.outputstream.OutputStreamInstrumentationModule BUT FAILURE WAS EXPECTED

FAILURE: Build failed with an exception.
```


However, if I then make `assertInverse=false`, I see the following error.
```
> Task :instrumentation:jaxrs-client-2.0:muzzle-AssertPass-io.dropwizard-dropwizard-client-4.0.0 FAILED
FAILED MUZZLE VALIDATION: io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyInstrumentationModule mismatches:
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyInstrumentationModule$ClientBuilder_build_Advice:83 Missing class javax.ws.rs.core.Configurable
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyInstrumentationModule$ClientBuilder_build_Advice:83 Missing class javax.ws.rs.client.Client
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientEntityInterceptor:65 Missing class javax.ws.rs.ext.ReaderInterceptorContext
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyCaptureFilter:0 Missing class javax.ws.rs.client.ClientResponseFilter
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyCaptureFilter:0 Missing class javax.ws.rs.client.ClientRequestFilter
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientEntityInterceptor:68 Missing class javax.ws.rs.core.MediaType
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientEntityInterceptor:0 Missing class javax.ws.rs.ext.ReaderInterceptor
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientEntityInterceptor:0 Missing class javax.ws.rs.ext.WriterInterceptor
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyCaptureFilter:51 Missing class javax.ws.rs.client.ClientRequestContext
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyCaptureFilter:65 Missing class javax.ws.rs.core.MultivaluedMap
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientEntityInterceptor:120 Missing class javax.ws.rs.ext.WriterInterceptorContext
-- io.opentelemetry.javaagent.instrumentation.hypertrace.jaxrs.v2_0.JaxrsClientBodyCaptureFilter:87 Missing class javax.ws.rs.client.ClientResponseContext

FAILURE: Build failed with an exception.
```

Reply:
As a compromise, I'll do the following:
```
versions = "(,4.0.0)"
assertInverse=false
skipVersions.add("4.0.0")
```

It has the net effect of:
1. passing the test.  =)
2. supporting everything up to (but not including) 4.0.0.
3. Will not fail when new versions come out.

Reply:
@tyroneyu We should also update the muzzle in jaxrs as well then. We can have `versions = "(,4.0.0)" and assertInverse=true` in apache-httpclient, and whatever changes needed in jaxrs muzzle as well.

Reply:
Oops, I made the change but didn't push that file. You are right and good catch.